lib/facter/util/resolvers/networking/networking.rb

```diff
@@ -65,7 +65,19 @@ def find_valid_binding(bindings)
           end
 
           def ignored_ip_address(addr)
-            addr.empty? || addr.start_with?('127.', '169.254.') || addr.start_with?('fe80') || addr.eql?('::1')
+            return true if addr.empty?
+
+            ip = IPAddr.new(addr)
```

Comment:
Can `addr` be invalid? If so, you'll want to rescue:

```ruby
ruby -r ipaddr -e "puts IPAddr.new('42')"
Traceback (most recent call last):
	3: from -e:1:in `<main>'
	2: from -e:1:in `new'
	1: from /home/josh/.rbenv/versions/2.7.6/lib/ruby/2.7.0/ipaddr.rb:586:in `initialize'
/home/josh/.rbenv/versions/2.7.6/lib/ruby/2.7.0/ipaddr.rb:651:in `in6_addr': invalid address (IPAddr::InvalidAddressError)
	3: from -e:1:in `<main>'
	2: from -e:1:in `new'
	1: from /home/josh/.rbenv/versions/2.7.6/lib/ruby/2.7.0/ipaddr.rb:557:in `initialize'
/home/josh/.rbenv/versions/2.7.6/lib/ruby/2.7.0/ipaddr.rb:598:in `rescue in initialize': invalid address: 42 (IPAddr::InvalidAddressError)
```

Reply:
> Can `addr` be invalid?

I don't think so: on line 18 when we build the binding, we already succeeded in parsing this address. Maybe it would make sense to only have `IPAddr` in `bindind` and build the `address` / `netmask` / `network` from the IPAddr later, but I didn't want to rework too much the code so decided to parse the addr a second time.